components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/IdentitySCIMManagerTest.java

```diff
@@ -0,0 +1,142 @@
+
+    @Mock
+    ClaimManager mockedClaimManager;
+
+    @Mock
+    UserStoreManager mockedUserStoreManager;
+
+    private SCIMConfigProcessor scimConfigProcessor;
+    private IdentitySCIMManager identitySCIMManager;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+        mockStatic(SCIMCommonUtils.class);
+        when(SCIMCommonUtils.getSCIMUserURL()).thenReturn("http://scimUserUrl:9443");
+
+        mockStatic(SCIMCommonComponentHolder.class);
+        when(SCIMCommonComponentHolder.getRealmService()).thenReturn(realmService);
+
+        scimConfigProcessor = SCIMConfigProcessor.getInstance();
+        String filePath = Paths
+                .get(System.getProperty("user.dir"), "src", "test", "resources", "charon-config-test.xml").toString();
+        scimConfigProcessor.buildConfigFromFile(filePath);
+        identitySCIMManager = IdentitySCIMManager.getInstance();
+
+        mockStatic(SCIMCommonComponentHolder.class);
+
+        when(realmService.getTenantManager()).thenReturn(mockedTenantManager);
+        when(mockedTenantManager.getTenantId(anyString())).thenReturn(-1234);
+        when(realmService.getTenantUserRealm(anyInt())).thenReturn(mockedUserRealm);
+
+        when(mockedUserRealm.getClaimManager()).thenReturn(mockedClaimManager);
+        when(mockedUserRealm.getUserStoreManager()).thenReturn(mockedUserStoreManager);
+        CommonTestUtils.initPrivilegedCarbonContext();
+    }
+
+    @ObjectFactory
+    public IObjectFactory getObjectFactory() {
+        return new org.powermock.modules.testng.PowerMockObjectFactory();
+    }
+
+    @Test
+    public void testGetInstance() throws Exception {
+        assertNotNull(identitySCIMManager, "Returning a null");
+        assertNotNull(identitySCIMManager, "Returning a null");
+    }
+
+    @Test
+    public void testGetEncoder() throws Exception {
+        assertNotNull(identitySCIMManager.getEncoder());
+    }
+
+    @Test
+    public void testGetUserManager() throws Exception {
+        when(SCIMCommonComponentHolder.getRealmService()).thenReturn(realmService);
+        UserManager userManager = identitySCIMManager.getUserManager();
+        assertNotNull(userManager);
+    }
+
+    @Test
+    public void testGetUserManagerWithException() throws Exception {
+        try {
+            when(SCIMCommonComponentHolder.getRealmService()).thenReturn(null);
+            identitySCIMManager.getUserManager();
+            fail("Expexted a CharonException, but did not throw");
+        } catch (CharonException e) {
+            assertNotNull(e);
+        }
+    }
+
+    @Test
+    public void testGetUserManagerWithException2() throws Exception {
+        try {
+            when(SCIMCommonComponentHolder.getRealmService()).thenReturn(realmService);
+            when(mockedTenantManager.getTenantId(anyString())).thenThrow(new UserStoreException());
+            identitySCIMManager.getUserManager();
+            fail("Expexted a CharonException, but did not throw");
```

Comment:
There is a typo in `Expexted`. Anyway better to use a message as below. 
`getUserManager() method should have thrown a CharonException.`